Update continuous-aggregates.md (#375)
timescale/docs.timescale.com-content: using-timescaledb/continuous-aggregates.md

**How it Works:** A *materialization background job* regularly takes
raw data from the hypertable and computes a partial aggregate that it
stores (materializes) in the continuous aggregate. When new data is 
inserted, updated, or deleted in the hypertable, the continuous aggregate
will automatically decide what data needs to be re-materialized and 
schedule a re-materialization to happen the next time the materialization 
job runs.

**Real-Time Aggregates** 

Real-time aggregates are a capability (first introduced in TimescaleDB 1.7)
whereby querying the *continuous aggregate view* will then compute an
up-to-date final aggregate result by combining the materialized
partial aggregate with recent data from the hypertable that has yet to
be materialized by the continuous aggregate. By combining raw and
materialized data in this way, one gets accurate and up-to-date
results while still enjoying the speedups of pre-computing a large
portion of the result.

> As an example, continuous aggregates made it really fast to get aggregate 
> answers by precomputing these values (such as the min/max/average value over
> each hour). This way, if you are collecting raw data every second, querying hourly
> data over the past week means reading 24 x 7 = 168 values from the database, as
> opposed to processing 60 x 60 x 24 x 7 = 604,800 values at query time.  But one
> limitation with continuous aggregates is that they don't incorporate the very
> latest data, _i.e._, since the last time the asynchronous aggregation job ran
> inside the database. So if you are generating hourly rollups, you might only
> run this job every hour.
> 
> But now with real-time aggregates, a single, simple query will combine your
> pre-computed hourly rollups with the raw data from the last
> hour, to always give you an up-to-date answer.  Now, instead of touching
> 604,800 rows of raw data, the query reads 167 pre-computed rows of
> hourly data and 3600 rows of raw secondly data, leading to significant
> performance improvements.

Reviewer: Sorry, my bad. This was text from my previous release notes.  I should have read more carefully :)

```suggestion
As an example, continuous aggregates _without_ this real-time capability
make it really fast to get aggregate 
answers by precomputing these values (such as the min/max/average value over
each hour). This way, if you are collecting raw data every second, querying hourly
data over the past week means reading 24 x 7 = 168 values from the database, as
opposed to processing 60 x 60 x 24 x 7 = 604,800 values at query time.  

But this type of continuous aggregate does not incorporate the very
latest data, _i.e._, since the last time the asynchronous aggregation job ran
inside the database. So if you are generating hourly rollups, you might only
run this materialization job every hour.

With real-time aggregates, a single, simple query will combine your
pre-computed hourly rollups with the raw data from the last
hour, to always give you an up-to-date answer.  Now, instead of touching
604,800 rows of raw data, the query reads 167 pre-computed rows of
hourly data and 3600 rows of raw secondly data, leading to significant
performance improvements.
```

Real-time aggregates are now the default behavior for any continuous aggregates.  
To revert to previous behavior, in which the query touches materialized data only 
and doesn't combine with the latest raw data, add the following parameter when 

> creating the continuous agggregate view
> 
> timescaledb.materialized_only=true

Reviewer: ```suggestion
creating the continuous aggregate view:

timescaledb.materialized_only=true
```

--> **Also** wrap `timescaledb.materialized_only=true` in triple backticks --> Sorry I couldn't do this in example, as markdown was interpreting it differently.

You can also use this in conjunction with the ALTER VIEW to turn this feature
on or off at any time.

> >:TIP: Upgrading continuous aggregates that were created in a version earlier that

Reviewer: ```suggestion
>:TIP: Upgrading continuous aggregates that were created in a version earlier than
```

raw data from the hypertable and computes a partial aggregate that it
stores (materializes) in the continuous aggregate.

> **Real-Time Aggregate** 

Reviewer: ```suggestion
**Real-Time Aggregates** 
```

Reviewer 2: I dont know maybe its too late for that but i really like to call it Real-Time Aggregation cause that works much better in conjunction with continuous aggregates

> Real-Time Aggregates are a concept that was introduced in TimescaleDB 1.7

Reviewer: ```suggestion
Real-time aggregates are a capability (first introduced in TimescaleDB 1.7)
```

> whereby Querying the *continuous aggregate view* will then compute an

Reviewer: Querying -> querying

Reviewer: Also, what about the following paragraph after this next paragraph (github interface not letting me add comment there:

```
As an example, continuous aggregates made it really fast to get aggregate 
answers by precomputing these values (such as the min/max/average value over
each hour). This way, if you are collecting raw data every second, querying hourly
data over the past week means reading 24 x 7 = 168 values from the database, as
opposed to processing 60 x 60 x 24 x 7 = 604,800 values at query time.  But one
limitation with continuous aggregates is that they don't incorporate the very
latest data, _i.e._, since the last time the asynchronous aggregation job ran
inside the database. So if you are generating hourly rollups, you might only
run this job every hour.

But now with real-time aggregates, a single, simple query will combine your
pre-computed hourly rollups with the raw data from the last
hour, to always give you an up-to-date answer.  Now, instead of touching
604,800 rows of raw data, the query reads 167 pre-computed rows of
hourly data and 3600 rows of raw secondly data, leading to significant
performance improvements.
```

Reviewer: Also the next paragraph "When new data"...doesn't seem to fit under RTA, and instead should be more top-level?

re-materialized and schedule a re-materialization to happen the next
time the materialization job runs.

> This is now the default behavior for any new continuous aggregate built
> after the 1.7 upgrade or for any new instance built with version 1.7.  
> To revert the continuous aggregate to the pre 1.7 behavior (that is not 
> including data that has not been materialized in your query) you can simply 
> add the following parameter at creation time:

Reviewer: ```suggestion
Real-time aggregates are now the default behavior for any continuous aggregates.  
To revert to previous behavior, in which the query touches materialized data only 
and doesn't combine with the latest raw data, add the following parameter when 
creating the continuous agggregate view
```

timescaledb.materialized_only=true

> You can also use this in conjunction with the ALTER VIEW to turn this feature
> on or off at any time, this is also the process for upgrading any continuous
> aggregates that were created prior to the 1.7 upgrade (setting this flag to FALSE).  
> Once you issue the ALTER VIEW and set the above parameter your continuous aggregate 
> will then use the real-time feature at query time.

Reviewer: ```suggestion
You can also use this in conjunction with the ALTER VIEW to turn this feature
on or off at any time.

>:TIP: Upgrading continuous aggregates that were created in a version earlier that
TimescaleDB 1.7 to use real-time aggregates should also ALTER the view to
set `timescaledb.materialized_only=false`.   Subsequent to this change, queries 
to the view will immediately use the real-time aggregate feature at query time.
```

>:TIP: Upgrading continuous aggregates that were created in a version earlier than
TimescaleDB 1.7 to use real-time aggregates should also ALTER the view to

> set ```timescaledb.materialized_only=false```.   Subsequent to this change, queries 

Reviewer 3: Don't think this is necessary

Reviewer 3: It doesn't affect how it parses.  If you want it in a non-inline code block, you need the backticks on separate lines
> \`\`\`
code here
\`\`\`

Reviewer 3: Looks fine inline though:
<img width="814" alt="Screen Shot 2020-04-22 at 12 14 39 PM" src="https://user-images.githubusercontent.com/12288273/80006442-e0cf6000-8492-11ea-87d7-555afeea1806.png">

Reviewer: Oops, yes, I meant this to be single ticks not three ticks.
